src/main/java/uk/gov/hmcts/reform/sscs/service/LetterUtils.java

```diff
@@ -197,4 +191,112 @@ public static Stream<Pair<String, YesNo>> buildOtherPartiesForReasonableAdjustme
         }
         return otherPartyReasonableAdjustmentList.stream();
     }
+
+    public static String getNameForSender(SscsCaseData sscsCaseData) {
+        if (nonNull(sscsCaseData.getOriginalSender())) {
+            final String originalSenderCode = sscsCaseData.getOriginalSender().getValue().getCode();
+            if (originalSenderCode.equalsIgnoreCase("appellant")) {
+                return sscsCaseData.getAppeal().getAppellant().getName().getFullNameNoTitle();
+            } else if (originalSenderCode.equalsIgnoreCase("representative")) {
+                return SendNotificationHelper.getRepSalutation(sscsCaseData.getAppeal().getRep(), false);
+            } else if (originalSenderCode.equalsIgnoreCase("jointParty")) {
+                return sscsCaseData.getJointParty().getName().getFullNameNoTitle();
+            } else {
+                return getOtherPartyName(sscsCaseData)
+                        .map(Name::getFullNameNoTitle).orElse("");
```

Comment:
Is this acceptable to return an empty string? I can see there is no difference then in returned value between null and non null original sender. Feels like if it is non null then there should certainly be a name, and if not, an exception thrown. What do you think?

Reply:
In case of the Event ActionFurtherEvidence and Post Hearing, the name will be non null, in all the other event the name is not required so it return "", not required to throw or handle the null pointer exception.